src/agent/onefuzz-agent/src/tasks/utils.rs

```diff
@@ -113,37 +114,23 @@ impl CheckNotify for tokio::sync::Notify {
 
 const DELAY: Duration = Duration::from_secs(10);
 
-pub fn file_uploader_monitor(synced_dir: SyncedDir) -> Result<impl Future> {
+async fn file_uploader_monitor(synced_dir: SyncedDir, event: Event) -> Result<()> {
+    let url = synced_dir.url.url();
+    let dir = synced_dir.path.clone();
     verbose!("monitoring {}", synced_dir.path.display());
 
-    let dir = synced_dir.path;
-    let url = synced_dir.url;
-
-    let mut monitor = DirectoryMonitor::new(&dir);
+    let mut monitor = DirectoryMonitor::new(dir);
     monitor.start()?;
+    let mut uploader = BlobUploader::new(url);
 
-    let monitor = monitor.for_each(move |item| {
-        verbose!("saw item = {}", item.display());
-
-        let url = url.clone();
-
-        async move {
-            event!(new_result; EventData::Path = item.display().to_string());
-
-            let mut uploader = onefuzz::uploader::BlobUploader::new(url.url());
-            let result = uploader.upload(item.clone()).await;
-
-            if let Err(err) = result {
-                error!("couldn't upload item = {}, error = {}", item.display(), err);
-            } else {
-                verbose!("uploaded item = {}", item.display());
-            }
+    while let Some(item) = monitor.next().await {
+        event!(event.clone(); EventData::Path = item.display().to_string());
```

Comment:
The event should be fired after the upload is successful

Reply:
I'd prefer to know that the event happened, even if the upload fails.